### PR TITLE
Redmine 12489/miseq designs

### PR DESCRIPTION
**Pull request by jmrein**

### CODE REVIEW

File: ; [hunk 1]
```diff
@@ -0,0 +1,477 @@
...
+sub _extract_data {
+    my ( $c, $datafile ) = @_;
+
+    open my $fh, '<:encoding(utf8)', $datafile->tempname or croak;
+    my $data = _read_file( $c, $fh );
+    close $fh or croak;
+    return $data;
+}
+
+sub generate_bwa_query_file {
```
Comment (PeterFCKeen) [inline ;]: BWA search could be handy as a util function

**Comment by PeterFCKeen** [comment]
You've accidentally committed a file named ;

**Comment by PeterFCKeen** [comment]
Code looks great. A few of the function could do with being moved into a util file since they'd be useful elsewhere

**Comment by jmrein** [comment]
@PeterFCKeen Any functions in particular?

**Comment by PeterFCKeen** [comment]
BWA related functions so the fasta creation one etc

**Pull request merged**